luiggymamani/hufflepuff-own-language
# Feature/ID-26_51_56/literal member statement

# Pull Request Template
#### Resolved task/US: Member (https://tree.taiga.io/project/santiago_c_saavedra-hufflepuff/task/51)
#### Resolved task/US: Literal (https://tree.taiga.io/project/santiago_c_saavedra-hufflepuff/task/26)
#### Resolved task/US: Statement (https://tree.taiga.io/project/santiago_c_saavedra-hufflepuff/task/56)
* * * *

## What's in this pull request?
I implemented the generators for literals, members and statement

## Type of change



## How Has This Been Tested?
1. Go to the main and paste this code:
You can uncomment or comment on what you want to test, for now it is possible to test the literals and members but the statement needs to be implemented in another US that my colleagues are working on.
```haskell
module Main (main) where

import AbstractSyntaxTree
import CodeGenerator

main :: IO ()
main = do
  -- Literals:
  let input = LTakeMemberAttribute (TakeMemberAttributeName "luiggy")
  --let input = LTakeMemberAttribute (TakeMemberAttributeRole "luiggy")
  --let input = LStringIdentifier (StringId "luiggy")
  --let input = LString (String "luiggy$%·")

  --Members:
  --let input = (Member (MemberValueName (String "luiggy")) (MemberValueRole (StringId "Developer")))
  --let input = (Member (MemberIdentifierName "luiggy") (MemberIdentifierRole "Dev"))
  let input = Member (MemberTakeName "luiggy") (MemberTakeRole "luiggy")

  -- Literals
  --let some = generateLiteral input
  -- Members
  let some = generateMember input

  writeFile "build/generated.js" some
  print some
```
3. Run the program
```shell
stack run
```
4. Go to the generated code js file to see the result:
```shell
➜ tree      
.
├── app
├── build
│   └── generated.js
├── src
└── test
```

## Checklist:

- [x] My code follows the style [guidelines](https://tree.taiga.io/project/santiago_c_saavedra-hufflepuff/wiki/methodology) of this project
